update oesign README.md; removed -z, -s options from oesign sign (#4218)
openenclave/openenclave: tools/oesign/README.md

To understand the full CLI of this tool, please refer to the `usage` printout below:

```
Usage: output/bin/oesign sign -e ENCLAVE_IMAGE [-c CONFIG_FILE] [SIGN_OPTIONS...]
Where:
    ENCLAVE_IMAGE -- path of an enclave image file
    CONFIG_FILE -- configuration file containing enclave properties

Description:

>     This option (1) injects runtime properties into an enclave image

Reviewer: ```suggestion
    This command (1) injects runtime properties into an enclave image
```

        Debug - whether debug mode should be enabled (1) or not (0) in enclave
        (default: 0)
        ProductID - the product identified number
        SecurityVersion - the security version number
        NumHeapPages - the number of heap pages for this enclave
        NumStackPages - the number of stack pages for this enclave
        NumTCS - the number of thread control structures for this enclave
        ExtendedProductID - a 128-bit globally unique identifier for the
        enclave if the 16-bit ProductID proves too restrictive  (SGX2 feature)
        FamilyID - product family identity to group different enclaves
        under a common identity (SGX2 feature)
        CapturePFGPExceptions - whether in-enclave exception handler should
        be enabled (1) or not (0) to capture #PF and #GP exceptions
        (SGX2 feature, default: 0)
        CreateZeroBaseEnclave - whether the enclave creation should be
        enabled (1) or not (0) with base address 0x0 (default: 0).
        StartAddress -  the enclave image address when CreateZeroBaseEnclave=1.
        The value should be a power of two and greater than
        /proc/sys/vm/mmap_min_addr

>     NOTE: If neither ExtendedProductID norFamilyID is set, Key Separation

Reviewer: ```suggestion
    NOTE: If neither ExtendedProductID nor FamilyID is set, Key Separation
```

    The properties are read from the CONFIG_FILE. They override any
    properties that were already defined inside the enclave image through

>     use of the OE_SET_ENCLAVE_SGX macro. If a property is not explicitly set,

Reviewer: ```suggestion
    use of OE_SET_ENCLAVE_SGX or OE_SET_ENCLAVE_SGX 2 macros. If a property is not explicitly set,
```

>     the default values override the definitions. These properties include:

Author: ```suggestion
    the default values override the definitions set by OE_SET_ENCLAVE_SGX macro. These properties include:
```

Reviewer: by OE_SET_ENCLAVE_SGX or OE_SET_ENCLAVE_SGX 2 macros

        Debug - whether enclave debug mode should be enabled (1) or not (0)
        ProductID - the product identified number
        SecurityVersion - the security version number
        NumHeapPages - the number of heap pages for this enclave
        NumStackPages - the number of stack pages for this enclave
        NumTCS - the number of thread control structures for this enclave
        ExtendedProductID - a 128-bit globally unique identifier for the
        enclave if the 16-bit ProductID proves too restrictive

>         FamilyID - Product family identity to group different enclaves

Author: 'product' in place of 'Product'

        under a common identity
        CapturePFGPExceptions - whether in-enclave exception handler should

>         be enabled (1) or not (0) to capture the #PF and #GP exceptions.

Reviewer: Simplified as follows.
```suggestion
        be enabled (1) or not (0) to capture the #PF and #GP exceptions (default: 0).
```

        Defaults to 0
        CreateZeroBaseEnclave - whether the enclave creation with base

>         address 0x0 should be enabled (1) or not (0).Defaults to 0

Author: Space between '.' and 'Default'

>         StartAddress -  the enclave image address when enclave base address

Reviewer: The default value for start address is not helpful.
```suggestion
        StartAddress - the enclave image address when CreateZeroBaseEnclave=1.
        The value should be power of two and greater than zero.
```
